### PR TITLE
Add Like Model

### PR DESCRIPTION
**Pull request by mdkrog**
Add Like Model
----

* A Like belongs to both user and photo
* A user can only like a particular photo once
* We are able to tell how many likes a photo has using a counter cache
and calling the method likes_count
* Create and Destroy are the only actions required in Likes controller
* Create a photo_card partial, that can be used universally wherever
a photo needs to be displayed. This partial also includes a like/unlike
button
* Add feature tests to simulate a user liking and unliking a photo

like

### CODE REVIEW

File: spec/factories.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 FactoryGirl.define do
+  factory :like do
+    photo_id 1
+    user_id 1
```
Comment (Jesus) [inline spec/factories.rb]: I would have used `association :photo` and `association :user`. That would create the associated models in our tests when necessary. See documentation for associations: https://github.com/thoughtbot/factory_girl/blob/master/GETTING_STARTED.md#associations

Or maybe you used `photo_id` and `user_id` for some reason I'm missing?

File: spec/features/photos/user_likes_a_photo_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require "rails_helper"
+
+feature "user likes a photo" do
+  scenario "user clicks the like button below a photo" do
+    user = create(:user)
+    photo = create(:photo)
+
+    sign_in(user)
+    visit photo_path(photo)
```
Comment (Jesus) [inline spec/features/photos/user_likes_a_photo_spec.rb]: You might like `visit photo_path(as: user)`.

**Comment by Jesus** [comment]
This commit is fantastic, good job!

Ok to merge.

File: spec/factories.rb (hunk 1 above)
Comment (mdkrog) [inline spec/factories.rb]: Great point. I'll make the change.

**Comment by Jesus** [comment]
That GIF is really useful. What did you use to create it?

**Pull request merged**